src/iperf_api.c

```diff
@@ -4088,24 +4088,46 @@ static int
 diskfile_send(struct iperf_stream *sp)
 {
     int r;
+    int buffer_left=0; // represents total data in buffer to be sent out 
```

Comment:
Thanks for the PR! I am wondering how this code would handle the situation where upon entry, the buffer has `blksize` bytes? In this case, `buffer_left` will be initialized to zero and, because we don't need to read any data, it'll stay zero until we get to the check in line 4123, at which point we declare we're done.

Would it make more sense to initialize `buffer_left` to `sp->diskfile_left` instead?

Reply:
You are right. To keep this function self-contained that corner case should be handled here.
It will imply updating line 4091 to initialize `buffer_left` to `sp->diskfile_left` as you suggested. Consequently the line 4098 will need to be updated to remove adding `sp->diskfile_left` again to the `buffer_left`. 

Lastly, for better clarity and code readability, it probably makes sense to replace conditions in lines 4105 and 4115 to use the alternative conditions already mentioned in the comments.